addons/hr_recruitment/hr_recruitment.py

```diff
@@ -309,9 +302,7 @@ def stage_find(self, cr, uid, applicants, job_id, domain=[], order='sequence', c
             job_ids.add(job_id)
         # OR all job_id and OR with applicant_default
         if job_ids:
-            search_domain = ['|', ('case_default', '=', True), ('job_ids', 'in', list(job_ids))]
-        else:
-            search_domain = [('case_default', '=', True)]
+            search_domain = [('job_ids', 'in', list(job_ids))]
```

Comment:
Considering you already defined the search_domain variable above, use += instead of =, so it is more robust to further changes. Otherwise, if we want to add something else in the domain before this line, we will have to change this line to have the += as keeping the current = will overwrite all previous changes to search_domain.